azurerm/internal/services/media/media_asset_filter_resource_test.go

```diff
@@ -0,0 +1,201 @@
+	}
+
+	return utils.Bool(resp.FilterProperties != nil), nil
+}
+
+func (r AssetFilterResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_asset_filter" "test" {
+  name                        = "Filter-1"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+  asset_name                  = azurerm_media_asset.test.name
+}
+
+`, r.template(data))
+}
+
+func (r AssetFilterResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_asset_filter" "import" {
+  name                        = azurerm_media_asset_filter.test.name
+  resource_group_name         = azurerm_media_asset_filter.test.resource_group_name
+  media_services_account_name = azurerm_media_asset_filter.test.media_services_account_name
+  asset_name                  = azurerm_media_asset.test.name
+}
+
+`, r.basic(data))
+}
+
+func (r AssetFilterResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_asset_filter" "test" {
+  name                        = "Filter-1"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+  asset_name                  = azurerm_media_asset.test.name
+  first_quality_bitrate       = 128000
+
+  presentation_time_range {
+    start_timestamp              = 0
+    end_timestamp                = 170000000
+    presentation_window_duration = 9223372036854775000
+    live_backoff_duration        = 0
+    timescale                    = 10000000
+    force_end_timestamp          = false
+  }
+
+  track {
+    selection {
+      property  = "Type"
+      operation = "Equal"
+      value     = "Audio"
+    }
+
+    selection {
+      property  = "Language"
+      operation = "NotEqual"
+      value     = "en"
+    }
+
+    selection {
+      property  = "FourCC"
+      operation = "NotEqual"
+      value     = "EC-3"
+    }
+  }
+
+
+  track {
+    selection {
+      property  = "Type"
+      operation = "Equal"
+      value     = "Video"
+    }
+
+    selection {
+      property  = "Bitrate"
+      operation = "Equal"
+      value     = "3000000-5000000"
+    }
+  }
+}
+
+`, r.template(data))
+}
+
+func (AssetFilterResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-media-%d"
+  location = "%s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctestsa1%s"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+}
+
+resource "azurerm_media_services_account" "test" {
+  name                = "acctestmsa%s"
```

Comment:
are capitals allowed for this property?

Reply:
No,Media Services Account name must contain only lowercase letters and numbers